core/src/main/java/org/apache/calcite/util/mapping/Mappings.java

```diff
@@ -1383,19 +1415,59 @@ public int getTargetCount() {
       return size;
     }
 
+    /**
+     * Returns the target that a source maps to.
+     *
+     * @param source source
+     * @return target
+     */
     public int getTarget(int source) {
+      if (source < 0 || source >= size) {
+        throw new IndexOutOfBoundsException("source #" + source
+            + " has no target in identity mapping of size " + size);
+      }
```

Comment:
size can be -1, in which case, we don't want exception.

Reply:
I missed this case. Thank you. Fixed.